src/gasman.c

```diff
@@ -1099,6 +1099,27 @@ static void MaybeMoveBags(void)
 
 #endif
 
+/****************************************************************************
+**
+*F  OutOfMemoryPanic(size) . . . . . . . . . .  handle running out of memory
+**
+*/
+
+static TooLargeAllocationHandlerFunc TooLargeAllocFunc;
+void SetTooLargeAllocationHandler(TooLargeAllocationHandlerFunc f)
+{
+    TooLargeAllocFunc = f;
+}
+
+void OutOfMemoryPanic(UInt size, const char * reason)
```

Comment:
Perhaps a batter name would be just `HandleOutOfMemory` or `SignalOutOfMemory` or just `OutOfMemory`, given that the function does *not* `Panic` in general, only if we really are out of memory...?